app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/receivers/GBMusicControlReceiver.java

```diff
@@ -73,6 +80,23 @@ public void onReceive(Context context, Intent intent) {
             Prefs prefs = GBApplication.getPrefs();
             String audioPlayer = prefs.getString("audio_player", "default");
 
+            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
+                MediaSessionManager mediaSessionManager =
+                        (MediaSessionManager) context.getSystemService(Context.MEDIA_SESSION_SERVICE);
+
+                List<MediaController> controllers = mediaSessionManager.getActiveSessions(
+                        new ComponentName(context, NotificationListener.class));
+                try {
+                    MediaController controller = controllers.get(0);
+                    audioPlayer = controller.getPackageName();
+                } catch (IndexOutOfBoundsException e) {
+                    System.err.println("IndexOutOfBoundsException: " + e.getMessage());
+                }
+            }
+
+            Log.d("keypress", musicCmd.toString());
+            Log.d("sent to", audioPlayer);
+
             long eventtime = SystemClock.uptimeMillis();
```

Comment:
I didn't try it, but it looks mostly good. Two things:
- could you please make all this a separate method `private String getAudioPlayer()`
- don't use System.*.println() and don't use android.util.Log. Instead use the slf4j LOG instance, and pass a message (and the exception, if any)

Thanks! :-)